### PR TITLE
Fix VPD page 0x80 serial number parsing to skip header bytes

### PR DESCRIPTION
**Pull request by Qubad786**
## Problem

Disk serial numbers from certain SCSI/SAS devices were incorrectly prefixed with a '%' character when read from the vpd_pg80 sysfs file. This occurred when:
- The regular /device/serial file didn't exist (common for SCSI/SAS devices)
- The disk serial number was exactly 37 bytes long (0x25 = '%' in ASCII)
- The VPD page 0x80 data was used as a fallback

The root cause was that the code incorrectly included the 4-byte VPD header in the serial string. When byte 3 (the length field) contained 0x25 (37 decimal, '%' in ASCII), it appeared as the first character of the serial.

## Solution

Fixed the VPD page 0x80 parsing to comply with SCSI specifications:
- Skip the 4-byte VPD header (bytes 0-3) when extracting serial data
- Use byte 3 as the single-byte length field (per SPC-6 § 7.6.10)
- Add defensive guard to prevent reading beyond buffer bounds
- Strip trailing null bytes and whitespace from the serial number
- Updated documentation to reference IBM and SCSI specifications

VPD page 0x80 structure:
- Byte 0: Peripheral qualifier & device type
- Byte 1: Page code (0x80)
- Byte 2: Reserved (0x00)
- Byte 3: Page length (max 255 bytes)
- Bytes 4+: ASCII serial number data

This ensures:
- Serial numbers match what udev reports as ID_SCSI_SERIAL
- No spurious '%' or other header bytes in serial numbers
- Robust handling of malformed VPD data
- Compliance with SCSI specifications

Added comprehensive unit tests including edge cases and defensive programming for devices that report incorrect lengths.

### CODE REVIEW

File: src/middlewared/middlewared/utils/disks_/disk_class.py [hunk 1]
```diff
@@ -126,11 +126,31 @@ def serial(self) -> str | None:
...
+                # IMPORTANT: VPD page 0x80 uses a single-byte length field (byte 3),
+                # unlike page 0x83 which uses a 16-bit big-endian length.
+                # Page 0x80 is limited to 255 bytes of serial data.
+                #
+                # Reference: IBM System Storage DS3500 Storage System SCSI Reference
+                # Section 4.7.5.12 INQUIRY Unit Serial Number Page 0x80:
+                # "The PAGE LENGTH field indicates the number of bytes that follow
+                # in the unit serial number page." (single byte at offset 3)
+                if len(serial) > 4:
+                    # Page length is a single byte value at byte 3
```
Comment (yocalebo) [inline src/middlewared/middlewared/utils/disks_/disk_class.py]: 1. The IBM specification is making reference to their specific storage system which isn't something that we should document in our code. Especially since IBM systems could be running powerPC architecture. Don't think it affects the SCSI specification too much but we shouldn't reference it either way.

File: src/middlewared/middlewared/utils/disks_/disk_class.py [hunk 2]
```diff
@@ -126,11 +126,31 @@ def serial(self) -> str | None:
         """The disk's serial number as reported by sysfs"""
         if not (serial := self.__opener(relative_path="device/serial")):
             if serial := self.__opener(relative_path="device/vpd_pg80", mode="rb"):
-                serial = (
-                    "".join(chr(b) if 32 <= b <= 126 else "\ufffd" for b in serial)
-                    .replace("\ufffd", "")
-                    .strip()
-                )
+                # VPD page 0x80 (Unit Serial Number) structure:
```
Comment (yocalebo) [inline src/middlewared/middlewared/utils/disks_/disk_class.py:129]: This layout is wrong according to SPC-7 rev 1. The PAGE LENGTH field is a two-byte field [Table 599]. This means it can be from 0, 65535 bytes in length.
